### DISCUSSION TITLE
Clone git repo onto R Studio server?

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1525] [Q&A]
@kubu4  I'm trying to clone the CEABiGR repository onto the R Studio server on Mox (need to use Mox because Raven did not have enough memory). I added the following code to the `singularity` definition file:

```
%post
    # Install additinoal system packages in container
    # Most are needed for R/RStudio dependencies
    apt -y update
    apt -y install libxml2 libz-dev libbz2-dev liblzma-dev libxtst6 git
    git clone --recursive https://github.com/sr320/ceabigr.git
```

I was able to create a container, but when I launch the R Studio server I have no idea where to find the files I cloned! Is there another way to clone these files/enable the Git panel in R Studio server? I know I can always `rsync` the files but it would be nice to push to the repo directory from Mox.

**Comment by kubu4** [comment]
You need to provide your desired location to clone the repo.

You'd likely want to add the following _above_ your git clone line:

`mkdir --parents $HOME/gitrepos && cd $_`

That will create a directory called `gitrepos` in your home directory and then change into that directory.

Thus, when the `git clone` command is executed, it will clone into `$HOME/gitrepos`.

In your version of the definition file above, you didn't change directories. So, my guess is that the git repo was cloned into the root directory of the container (i.e. `cd /`).
